ui/apps/platform/src/Components/CompoundSearchFilter/components/CompoundSearchFilterInputField.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+
+import { SearchInput, SelectOption } from '@patternfly/react-core';
+import { SelectedEntity } from './EntitySelector';
+import { SelectedAttribute } from './AttributeSelector';
+import {
+    CompoundSearchFilterConfig,
+    SearchFilterAttribute,
+    SelectSearchFilterAttribute,
+} from '../types';
+
+import CheckboxSelect from './CheckboxSelect';
+
+export type InputFieldValue = string | number | undefined | string[];
+export type InputFieldOnChange = (value: InputFieldValue) => void;
+
+export type CompoundSearchFilterInputFieldProps = {
+    selectedEntity: SelectedEntity;
+    selectedAttribute: SelectedAttribute;
+    value: InputFieldValue;
+    onSearch: InputFieldOnChange;
+    onChange: InputFieldOnChange;
+    config: Partial<CompoundSearchFilterConfig>;
+};
+
+function isSelectType(
+    attributeObject: SearchFilterAttribute
+): attributeObject is SelectSearchFilterAttribute {
+    return attributeObject.inputType === 'select';
+}
+
+function CompoundSearchFilterInputField({
+    selectedEntity,
+    selectedAttribute,
+    value,
+    onSearch,
+    onChange,
+    config,
+}: CompoundSearchFilterInputFieldProps) {
+    if (!selectedEntity || !selectedAttribute) {
+        return null;
+    }
+
+    const attributeObject: SearchFilterAttribute =
+        config[selectedEntity]?.attributes[selectedAttribute];
+
+    if (!attributeObject) {
+        return null;
+    }
+
+    if (attributeObject.inputType === 'text') {
+        const textLabel = `Filter results by ${attributeObject.filterChipLabel.toLowerCase()}`;
+        return (
+            <SearchInput
+                aria-label={textLabel}
+                placeholder={textLabel}
+                value={value as string}
+                onChange={(_event, _value) => onChange(_value)}
+                onSearch={(_event, _value) => onSearch(_value)}
+                onClear={() => onChange('')}
+            />
+        );
+    }
+    if (isSelectType(attributeObject)) {
+        const attributeLabel = attributeObject.displayName.toLowerCase();
+        const selection = value as string[];
+        const selectOptions = attributeObject.inputProps.options;
+
+        return (
+            <CheckboxSelect
```

Comment:
Thought on styling after looking at test screenshots.

Would it be better to have the input element extend to take up the remaining width of its container? It looks like the <SearchInput> behaves this way, but `<CheckboxSelect>` does not. Would that look better, or would it look strange when used in the context of a full page?

Reply:
Good question. I think maybe it would be best to keep them all the same width, but not sure how reasonable that is. Personally, the text input looks fine when it takes up the full width, but the select component might look a bit weird. We can see if UX has any thoughts on this. (cc: @alanonthegit @zhenpesky )

Reply:
I think it is fine to extend the text input since we are keeping the consistency of extending the width of a text-input submit-button in VM filters and Listening endpoints. The "submit button with text input" in this case has an individual look while having 2 regular dropdowns next to it, they serve different purposes/behaviors, so it is fine to extend it. 
However, the "node component", "source" and "filter by source" use the same dropdown component and behavior, it would be odd to extend the width as this leads to a different visual perception because all 3 dropdowns are actually the same component behavior to select an item (without text input). The case I found how the dropdown should be extended to take up the remaining width of the container is when there is only 1 dropdown in a row (an example would be inside the "configure report" wizard in VM reports).